### DISCUSSION TITLE
V7.3 Portable Error

### DISCUSSION THREAD

**Discussion by Androxilogin** [sanni/cartreader#293] [Coding Corner]
Receiving this message:

Arduino: 1.8.16 (Windows 10), Board: "Arduino Mega or Mega 2560, ATmega2560 (Mega 2560)"

In file included from C:\Program Files\Arduino\portable\sketchbook\libraries\SdFat\src\ExFatLib\upcase.cpp:25:0:

C:\Program Files\Arduino\portable\sketchbook\libraries\SdFat\src\ExFatLib\upcase.h:31:25: error: 'ExChar16_t' does not name a type; did you mean 'char16_t'?

                   const ExChar16_t* name, size_t offset, size_t n);

                         ^~~~~~~~~~

                         char16_t

C:\Program Files\Arduino\portable\sketchbook\libraries\SdFat\src\ExFatLib\upcase.h:33:30: error: 'ExChar16_t' does not name a type; did you mean 'char16_t'?

 uint16_t exFatHashName(const ExChar16_t* name, size_t n, uint16_t hash);

                              ^~~~~~~~~~

                              char16_t

C:\Program Files\Arduino\portable\sketchbook\libraries\SdFat\src\ExFatLib\upcase.cpp:207:25: error: 'ExChar16_t' does not name a type; did you mean 'char16_t'?

                   const ExChar16_t* name, size_t offset, size_t n) {

                         ^~~~~~~~~~

                         char16_t

C:\Program Files\Arduino\portable\sketchbook\libraries\SdFat\src\ExFatLib\upcase.cpp:230:30: error: 'ExChar16_t' does not name a type; did you mean 'char16_t'?

 uint16_t exFatHashName(const ExChar16_t* name, size_t n, uint16_t hash) {

                              ^~~~~~~~~~

                              char16_t

exit status 1

Error compiling for board Arduino Mega or Mega 2560.

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

I'm assuming it's just a typo but thought I'd let it be known.

**Comment by sanni** [comment]
I can't get this error to show for me. I do find results using google with the same error though. Certainly interesting, seems like a bug in the SD library that only happens for some.

**Reply by Androxilogin** [reply]
I'll give it a try on my laptop whenever I get a chance and see if the problem persists.

**Comment by Androxilogin** [comment]
Hate to double-post but the issue seemed to have resolved itself by not placing the extracted files in Program Files.

**Comment by sanni** [comment]
Cool, I'll add this advice to the download page. Thanks for finding the cause of the issues and reporting back.
